### ISSUE TITLE
Feature: support copy files from one stage to other stage

### ISSUE THREAD

**Issue by wubx** [datafuselabs/databend#12200]
**Summary**

Databend is use object storage heavily.  Maybe support copy file from one stage to other stage is better.
```
copy files from @mystage1/path/ to @mystage2;
copy files from @mystage1/path/f1.csv to @mystage2/archive/[f1.csv];
```

**Comment by rahul3002** [comment]
If this issue is open, I would love to sink my teeth into it!

**Comment by wubx** [comment]
Good, If you have any questions, please comment here.

**Comment by rahul3002** [comment]
Could you tell me where the files are located and which files need to be copied to another location?

where is another location

so i can head it on.

**Comment by wubx** [comment]
This feture copy files from stage to stage
About stage you find in: https://databend.rs/doc/sql-commands/ddl/stage/ddl-create-stage

**Comment by thomas-k-cameron** [comment]
Hey @wubx,

So I was interested in the issue so I decided look into it.

I wrote a function for copying from one to another, but I have few questions.

- Where can I find the code for receiving user inputs?
- Am I correct to believe that I need to add a new parser that accepts inputs like (`COPY FILES FROM @mystage1/path/ TO @mystage2`), if so, where should I write the parsing logic?

Also, I made a draft pull request so that we can look at my implementation, and be on the same pages.
https://github.com/datafuselabs/databend/pull/12400
